spec/models/user_spec.rb

```diff
@@ -1814,4 +1814,29 @@ def filter_by(method)
 
   end
 
+  describe '#match_title_to_primary_group_changes' do
+    let(:previous_primary_group) { Fabricate(:group, title: 'Previous', users: [user]) }
+    let(:new_primary_group) { Fabricate(:group, title: 'New', users: [user]) }
+
+    before { user.update(primary_group: previous_primary_group) }
+
+    def change_primary_group
+      user.update(primary_group: new_primary_group)
+    end
+
+    it "updates user's title when it matches the previous primary group" do
+      user.update(title: 'Previous')
```

Comment:
lets merge all these 3 specs if we can cleanly? Overall the change looks good to me. 👍